dvc/remote/base.py

```diff
@@ -142,23 +149,32 @@ def cache(self):
     def get_file_checksum(self, path_info):
         raise NotImplementedError
 
-    def _collect_dir(self, path_info):
-        dir_info = {}
+    def _get_file_checksum_future(self, file_info, executor):
+        checksum = self.state.get(file_info)
+        if checksum:
+            checksum = as_future(checksum)
```

Comment:
There is not much sense in wrapping it specifically for this purpose. You could just check isinstance Future when trying to retrieve the result.

Reply:
Initially I agreed, but the problem is that before wrapping it in progress bar (on large dir) we wrap checksum into as_completed, which requires Futures. We can solve it by putting checksums into different container than futures, what do you think?

Reply:
Great point! I guess there is no reason to wrap it into anything else in that case.